internal/controllers/provisioningrequest_hwprovision.go

```diff
@@ -429,6 +429,12 @@ func (t *provisioningRequestReconcilerTask) handleRenderHardwareTemplate(ctx con
 	nodePool.ObjectMeta.Name = clusterInstance.GetName()
 	nodePool.ObjectMeta.Namespace = utils.GetHwMgrPluginNS()
 
+	// Add extensions from HWtemplate
+	extensions, err := utils.ExtractMatchingInput(t.object.Spec.TemplateParameters.Raw, utils.HwTemplateExtensions)
```

Comment:
I assumed the extensions field would be sourced from the hardware template rather than the ClusterTemplate. Can you clarify if this is the intended behavior?

Reply:
It should be coming from the hwtemplate